draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -848,6 +848,152 @@ Content-Type: text/plain
+   binding "http" identifiers to an origin server for consistent resolution
+   within a trusted environment.
+</t>
+<t>
+   If the host identifier is provided as an IP address, the origin server is
+   the listener (if any) on the indicated TCP port at that IP address.
+   If host is a registered name, the registered name is an indirect identifier
+   for use with a name resolution service, such as DNS, to find an address for
+   an appropriate origin server.
+</t>
+<t>
+   Access to "http" services is not limited to the authority mechanism.
+   For example, the Alt-Svc header field <xref target="RFC7838"/> allows an
+   origin server to explicitly alias its own namespace to an alternative
+   service at a different origin. Access to "http" identified resources might
+   also be provided by other protocols outside the scope of this document.
+</t>
+<t>
+   The "https" scheme, in contrast, associates authority based on the ability
+   of a server to use the private key associated with a certificate that
+   the client considers to be trustworthy for the identified host.
+   If a server presents a certificate that verifiably applies to the host,
+   along with proof that it controls the private key, then a client will
+   accept a secured connection to that server as being authoritative for all
+   origins with the same host. However, each origin remains distinct (from
+   origins with a different scheme or port) unless explicitly aliased by an
+   origin server verified to be under the same authority.
+</t>
+<t>
+   What the client is relying on for "https" authority is that there exists
+   some chain of trust (possibly configured by exception) that verifies the
+   certificate provided. Hence, the authority is defined only by the host,
+   the port impacts which certificate is chosen (assuming that certificate
+   matches the host identifier), and the client trusts that match according
+   to the secure handshake and certificate validation described in
+   <xref target="https.initiation"/>.
+</t>
+<t>
+   Note that the "https" scheme does not rely on TCP and the port number for
+   establishing authority, even though it provides for use of HTTP over TLS
+   over TCP via that port as the default mechanism for obtaining a
+   certificate that can be used for verifying authority and providing access
+   to the origin's resources. The peers establishing a secure connection
+   cannot trust transport parameters outside the secured channel, but the
+   port number remains useful (beyond the default mechanism) as an identifier
+   to distinguish different origin namespaces on the same host. The port
+   identifier is therefore applicable even when the protocols being used for
+   access do not include TCP.
+</t>
+<t>
+   In HTTP/1.1 and earlier, the only URIs for which the client will assign
+   authority are those that contain a TLS server_name extension matching the
+   origin's host.
```

Comment:
I know what you're trying to say, but tying this to the SNI doesn't work.  It's legitimate for a client not to send that extension.

Reply:
You've lost me.  I thought this (from Martin) was talking about the TLS certificate matching the same name as the origin host. What is the right way to say that?

Reply:
> In HTTP/1.1 and earlier, the only URIs for which a client will attribute authority to a server are those for which a TLS connection was specifically established toward the origin's host. Only the characteristics of the connection establishment and certificate are used.  For a host that is a domain name, that name is included in the TLS server_name extension <xref target="RFC6066"/> and &MUST; also appear as either the CN field of the certificate subject or as a dNSName in the subjectAltName field of the certificate (see <xref target="section defining this..."/>)  A host that is an IP address &MUST; appear in the subjectAltName of the certificate.

Reply:
Thanks.  I've tweaked it a bit to make these requirements on the client and added a ref to RFC6125.